src/jit/morph.cpp

```diff
@@ -11221,15 +11221,20 @@ CM_OVF_OP :
             }
             else
             {
-                /* The left operand is worthless, throw it away */
-                if (lvaLocalVarRefCounted)
+                // If this is an intrinsic that represents the object.GetType(), it can throw an NullReferenceException.
```

Comment:
This is not the right way to model the side effect of obj.GetType().  Since importer is setting GTF_EXCEPT flag, we should modify GenTree::OperMayThrow() definition to return true in case of GetType() intrinsic.